src/service/vaults.ts

```diff
@@ -88,7 +95,14 @@ type VaultSubscribers = {
  * Ex. `asset: 'published.vaultFactory.manager0'` -> `'manager0'`.
  */
 const getManagerIdFromSubscribers = (subscribers: VaultSubscribers) =>
-  subscribers.vault.split('.').find(node => node.includes('manager'));
+  subscribers.vault
+    .split('.')
+    .find(node => node.startsWith('manager') && !node.startsWith('managers'));
+
+const lastNode = (path: string) => {
```

Comment:
Similarly to the "node" comment above. We're not getting out a node object, just a segment of the `path` param.
```suggestion
const lastSegment = (path: string) => {
```

Reply:
Done